dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -455,6 +460,370 @@ def traverse(mtree_node):
+
+        # Check for uses in state
+        access_nodes = set()
+        for e in graph.in_edges(nsdfg):
+            src = graph.memlet_path(e)[0].src
+            if isinstance(src, nodes.AccessNode) and graph.in_degree(src) == 0:
+                access_nodes.add(src)
+        for e in graph.out_edges(nsdfg):
+            dst = graph.memlet_path(e)[-1].dst
+            if isinstance(dst, nodes.AccessNode) and graph.out_degree(dst) == 0:
+                access_nodes.add(dst)
+        for node in graph.data_nodes():
+            if node.data in candidates and node not in access_nodes:
+                del candidates[node.data]
+
+        return candidates
+
+    @staticmethod
+    def can_be_applied(graph: SDFGState,
+                       candidate: Dict[transformation.PatternNode, int],
+                       expr_index: int,
+                       sdfg: SDFG,
+                       strict: bool = False):
+        nsdfg = graph.node(candidate[InlineTransients.nsdfg])
+
+        # Not every schedule is supported
+        if strict:
+            if nsdfg.schedule not in (dtypes.ScheduleType.Default,
+                                      dtypes.ScheduleType.Sequential,
+                                      dtypes.ScheduleType.CPU_Multicore,
+                                      dtypes.ScheduleType.GPU_Device):
+                return False
+
+        candidates = InlineTransients._candidates(sdfg, graph, nsdfg)
+        return len(candidates) > 0
+
+    @staticmethod
+    def match_to_str(graph, candidate):
+        return graph.label
+
+    def apply(self, sdfg):
+        state: SDFGState = sdfg.nodes()[self.state_id]
+        nsdfg_node: nodes.NestedSDFG = self.nsdfg(sdfg)
+        nsdfg: SDFG = nsdfg_node.sdfg
+        toremove = InlineTransients._candidates(sdfg, state, nsdfg_node)
+
+        for dname, cname in toremove.items():
+            # Make nested SDFG data descriptors transient
+            nsdfg.arrays[cname].transient = True
+
+            # Remove connectors from node
+            nsdfg_node.remove_in_connector(cname)
+            nsdfg_node.remove_out_connector(cname)
+
+            # Remove data descriptor from outer SDFG
+            del sdfg.arrays[dname]
+
+        # Remove edges from outer SDFG
+        for e in state.in_edges(nsdfg_node):
+            if e.data.data not in toremove:
+                continue
+            tree = state.memlet_tree(e)
+            for te in tree:
+                state.remove_edge_and_connectors(te)
+            # Remove newly isolated node
+            state.remove_node(tree.root().edge.src)
+
+        for e in state.out_edges(nsdfg_node):
+            if e.data.data not in toremove:
+                continue
+            tree = state.memlet_tree(e)
+            for te in tree:
+                state.remove_edge_and_connectors(te)
+            # Remove newly isolated node
+            state.remove_node(tree.root().edge.dst)
+
+
+class ASTRefiner(ast.NodeTransformer):
```

Comment:
It's not immediately obvious what this is, maybe mention that it's a utility for use in `RefineNestedAccess`, so someone can look up what this is for there.